### DISCUSSION TITLE
User permissions in /wp-content/reactpress/apps

### DISCUSSION THREAD

**Discussion by indymaat** [rockiger/reactpress#43] [Q&A] [answered]
# Hi community!
I am relatively new to Linux, WordPress and this plugin. However I do have good understanding of React.
I followed [this guide](https://ubuntu.com/tutorials/install-and-configure-wordpress) to install WordPress on my machine and it is working fine. However, [a part of the installation is done using a `www-data` user](https://ubuntu.com/tutorials/install-and-configure-wordpress#6-configure-wordpress-to-connect-to-the-database):
```
6. Configure WordPress to connect to the database

Now, let’s configure WordPress to use this database. First, copy the sample configuration file to wp-config.php:

sudo -u www-data cp /srv/www/wordpress/wp-config-sample.php /srv/www/wordpress/wp-config.php

Next, set the database credentials in the configuration file (do not replace database_name_here or username_here in the commands below. Do replace <your-password> with your database password.):

sudo -u www-data sed -i 's/database_name_here/wordpress/' /srv/www/wordpress/wp-config.php
sudo -u www-data sed -i 's/username_here/wordpress/' /srv/www/wordpress/wp-config.php
sudo -u www-data sed -i 's/password_here/<your-password>/' /srv/www/wordpress/wp-config.php

Finally, in a terminal session open the configuration file in nano:

sudo -u www-data nano /srv/www/wordpress/wp-config.php

```

If I now want to create a new react app in the directory of ReactPress i get the following:
```
$ sudo npx create-react-app my-reactpress-app
[sudo] password for indy: 
sudo: npx: command not found
```
I found out it is because root does not have node installed, and I read that installing node for root is not a good idea.
I tried it using `sudo -i` and use my `node` installation in the current session, although it works. Every code modification now requires sudo priviliges...
```
$ sudo -i
root@indy-thinkpad:~# export PATH=$PATH:/home/indy/.nvm/versions/node/v18.15.0/bin
root@indy-thinkpad:~# npx create-react-app /srv/www/wordpress/wp-c
wp-comments-post.php  wp-config-sample.php  wp-cron.php
wp-config.php         wp-content/           
root@indy-thinkpad:~# npx create-react-app /srv/www/wordpress/wp-content/reactpress/apps/my-reactpress-app

Creating a new React app in /srv/www/wordpress/wp-content/reactpress/apps/my-reactpress-app.
...
```

What I am not understanding, and what is the proper way of addressing this issue?

Thanks in advance.

**Comment by rockiger** [comment, marked as answer]
Hi @indymaat,

I would advice you to not install WordPress this way for local development. This is a server installation for a live Ubuntu server.  If I were you I would use [LocalWP](https://localwp.com/) for development.

I have written an [extensive tutorial on how to create your first ReactPress app](https://rockiger.com/en/reactpress/reactpress-tutorial/). It also includes the creation of a WordPress site. 

Hope this helps.

**Reply by indymaat** [reply]
Thanks @rockiger, I'll have a look at the extensive tutorial. Great!
